src/aff_expr.jl

```diff
@@ -809,3 +809,74 @@ moi_function(a::Vector{<:GenericAffExpr}) = MOI.VectorAffineFunction(a)
 function moi_function_type(::Type{<:Vector{<:GenericAffExpr{T}}}) where {T}
     return MOI.VectorAffineFunction{T}
 end
+
+"""
+    _eval_as_variable(f::F, x::GenericAffExpr, args...) where {F}
+
+In many cases, `@variable` can return a `GenericAffExpr` instead of a
+`GenericVariableRef`. This is particularly the case for complex-valued
+expressions. To make common operatons like `lower_bound(x)` work, we should
+forward the method if and only if `x` is convertable to a `GenericVariableRef`.
+"""
+function _eval_as_variable(f::F, x::GenericAffExpr, args...) where {F}
+    if length(x.terms) != 1
+        error(
+            "Cannot call $f with $x because it is not a real-valued affine " *
+            "expression of one variable.",
+        )
+    end
+    variable, coefficient = first(x.terms)
+    if !isone(coefficient)
+        error(
+            "Cannot call $f with $x because it is not a real-valued affine " *
+            "expression of one variable with a coefficient of `+1`.",
+        )
+    end
+    return f(variable, args...)
+end
+
+function _eval_as_variable(
+    f::F,
+    x::GenericAffExpr{T},
+    args...,
+) where {F,T<:Complex}
```

Comment:
Maybe we can remove this method so that `set_start_value(one(ComplexF64)` * x, 1.0)` works. This would be useful for the diagonal of hermitian matrices which are real